### ISSUE TITLE
ratio plot

### ISSUE THREAD

**Issue by lukasheinrich** [scikit-hep/mplhep#153]
Hi,

for some of my work I created a somehwat drop-in-ish replacement for `plt.subplots` but for ratio plots.. is this something that'd be interesting for this package?

cc @kratsg @matthewfeickert 

<img width="701" alt="screenshot" src="https://user-images.githubusercontent.com/2318083/83695007-87ae2c80-a5f9-11ea-867d-dadeda591d08.png">

**Comment by lukasheinrich** [comment]
fwiw it seems to play nice with mplhep

<img width="621" alt="screenshot" src="https://user-images.githubusercontent.com/2318083/83696416-ba0d5900-a5fc-11ea-8df7-c7e5053c7865.png">

**Comment by andrzejnovak** [comment]
That's kinda neat. I collect similar helper functions in `mplhep.plot` as well, though they're not well documented yet. A PR is welcome! One thing that feels kinda weird is having the subplot as a dict with `main` and `aux` could it be a tuple instead?

**Comment by andrzejnovak** [comment]
hmm or better yet, two arrays like `fig, ax, subax = ...` then the shape is clear and you can iterate over both with zip

**Comment by nsmith-** [comment]
There is a ratioplot utility in coffea that might be able to be ported over, as was done for the others: https://coffeateam.github.io/coffea/api/coffea.hist.plotratio.html

**Comment by acampove** [comment]
The problem with this is that the type of plots that ATLAS, LHCb, etc will accept for publication should look more like:

![image](https://user-images.githubusercontent.com/14037062/120888141-5495e300-c629-11eb-988b-7e54fc1f00fb.png)

I won't use anything that looks radically different because at the end the collaboration will make us change it so that it looks like the standard style. For which I would end up using the old ROOT way of doing things anyway.

**Comment by andrzejnovak** [comment]
@acampove I am not sure I follow. This issue is about automating layout for multiple subplots. The styling is separate and fairly straightforward.

```
import matplotlib.pyplot as plt
import mplhep as hep
import numpy as np

hists = [np.histogram(np.random.normal(i, 2, 10000), bins=np.linspace(0, 10, 50)) for i in [4, 5, 6]]

hep.histplot(hists[:2])
hep.histplot(hists[2], histtype='errorbar')
```
![image](https://user-images.githubusercontent.com/13226500/120890191-5481ed00-c601-11eb-9bcd-e547059c137d.png)

```
hep.style.use('ATLAS')
hep.histplot(hists[:2], color=['black', 'red'], label=['mc1', 'mc2'])
hep.histplot(hists[2], histtype='errorbar', color='k', yerr=True, label='Data')
plt.legend()
```

![image](https://user-images.githubusercontent.com/13226500/120890220-7e3b1400-c601-11eb-967f-342342faebc2.png)

**Comment by lukasheinrich** [comment]
yeah this was just about being able to transparenttly arrange a set of ratio plots similar to subplots..

**Comment by acampove** [comment]
@andrzejnovak Yes, I did not know that switching from that matplotlib style to an ATLAS style was that simple and that the layout was so easy to change.

**Comment by radiradev** [comment]
Hi @lukasheinrich could you share the code for this? I have been looking for something similar.

**Comment by kratsg** [comment]
It'll likely be code that comes from plothist -- https://plothist.readthedocs.io/en/latest/ -- given it is being merged with mplhep.

**Comment by matthewfeickert** [comment]
@zlmarshall there already is `plot_ratio` in `hist` (c.f. https://hist.readthedocs.io/en/latest/user-guide/notebooks/Plots.html#Via-Plot-Ratio) but that might be slightly different from what you're going for(?). Let us know.

**Comment by cyrraz** [comment]
> Thanks [@kratsg](https://github.com/kratsg) ! When you say 'being merged' — is there a timeline for that already?
> 
> Best, Zach

Hi Zach,

To answer our question, we want to be done with the merge this autumn. An important PR from `plothist` to `mplhep` was merged last week and we are currently working on the next one.

Best,
Cyrille

**Comment by zlmarshall** [comment]
That's great news, thanks Cyrille.  Good luck with the merge!

**Comment by jonas-eschle** [comment]
Just to add to this, @zlmarshall , feel free to chip in on any design/implementation decision and help reviewing our PR (will tag you if I remember, otherwise just do it yourselves). Input from users is always highly appreciated!
